commit 4ca6c1a060943161c6c2ce09e02ed58a69669cfe
Author: Peter Zijlstra <peterz@infradead.org>
Date:   Tue Apr 21 12:09:13 2020 +0200

    sched,acpi_pad: Convert to sched_set_fifo*()
    
    Because SCHED_FIFO is a broken scheduler model (see previous patches)
    take away the priority field, the kernel can't possibly make an
    informed decision.
    
    In this case, use fifo_low, because it only cares about being above
    SCHED_NORMAL. Effectively no change in behaviour.
    
    XXX: this driver is still complete crap; why isn't it using proper
    idle injection or at the very least play_idle() ?
    
    Cc: rafael.j.wysocki@intel.com
    Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
    Reviewed-by: Ingo Molnar <mingo@kernel.org>

diff --git a/drivers/acpi/acpi_pad.c b/drivers/acpi/acpi_pad.c
index e7dc0133f817..0ad9f2059e6d 100644
--- a/drivers/acpi/acpi_pad.c
+++ b/drivers/acpi/acpi_pad.c
@@ -136,12 +136,11 @@ static unsigned int idle_pct = 5; /* percentage */
 static unsigned int round_robin_time = 1; /* second */
 static int power_saving_thread(void *data)
 {
-	struct sched_param param = {.sched_priority = 1};
 	int do_sleep;
 	unsigned int tsk_index = (unsigned long)data;
 	u64 last_jiffies = 0;
 
-	sched_setscheduler(current, SCHED_RR, &param);
+	sched_set_fifo_low(current);
 
 	while (!kthread_should_stop()) {
 		unsigned long expire_time;